hdfs/ext/avro/__init__.py

```diff
@@ -64,18 +65,22 @@ def _infer_schema(obj):
       raise ValueError('Cannot infer type of empty array.')
     return {
       'type': 'array',
-      'items': _infer_schema(obj[0])
+      'items': _infer_schema(obj[0], record_name=record_name)
     }
   elif isinstance(obj, dict):
     if not obj:
       raise ValueError('Cannot infer type of empty record.')
-    return {
+    nested_record_name = u'Nested' + record_name if record_name else None
```

Comment:
Yeah I assumed that records have the same type at a given level. I did not have the chance to test it so I do not know how Avro reacts in this case.